Deploy

fejiroofficial opened the pull request:
**What does this PR do?
This a pull request for configuring eslint, removing an if statement from code block and for changing the name of a variable.

**Description of Task to be completed?**
Have the following
- have a proper indentation on test files(question.spec.js & answer.spec.js)
- have the if statement removed from `GET /api/v1/questions` controller function.

**What are the relevant pivotal tracker stories?**
#159951517

fejiroofficial committed:
Update .codeclimate.yml

fejiroofficial committed:
Heroku H10 Error
This is a commit for addressing heroku deployment error

fejiroofficial committed:
ch- uninstalling body-parser middle-ware
This is a commit to
-uninstall babel
-modifying run script start-dev to build
-remove code climate's test reporter badge.

[Fixes #159912524]

fejiroofficial committed:
ft(feedback implentation): Renaming question object properties.
This is a commit to change an object property
-The property title is now questionTitle.
-The property body is now questionDescription.

[Fixes #159920137]

fejiroofficial committed:
ft(feedback implentation): removing an if statement.
This is a commit that removes an
 if statement from the getAllQuestions controller function.
If statement not needed.

[Fixes  #159920875]

fejiroofficial committed:
ch(eslint configuration)Indentation)
This is a commit for configuring indentation rule to a 2-line spacing

[Fixes 159951517]

Billmike commented on .eslintrc:
```diff
@@ -5,26 +5,23 @@
-        "eslint:recommended",
-        "airbnb"
-    ],
+    "extends": "eslint:recommended",
```
Your `eslint` config doesn't extend `airbnb` style guide. Any reason why?

Billmike commented on .slugignore:
```diff
@@ -0,0 +1 @@
+yarn.lock
```
Add a new line here

Billmike commented on .eslintrc:
```diff
@@ -5,26 +5,23 @@
         "quotes": [
             "error",
-            "double"
+            "single"
```
With `airbnb`, preferred quotes are double, not single.

fejiroofficial replied:
okay

Billmike commented on test/controller/answers.spec.js:
```diff
@@ -7,57 +7,57 @@ import { allQuestions, answers } from '../../datastore/questions';
-						done();
-					});
-			});
+  // test answer controller
```
No need for the in-line comments.

fejiroofficial replied:
noted

Billmike commented on test/controller/answers.spec.js:
```diff
@@ -7,57 +7,57 @@ import { allQuestions, answers } from '../../datastore/questions';
+    
+    describe('when invalid data is passed', () => {
+      it('should not create an answer resource', (done) => {
+        const invalidAnswer = Object.assign({}, validAnswer, {
```
You could re-write this to be like this:
```javascript
const invalidAnswer = { ...validAnswer, answer: '' };
```
This is cleaner, and achieves the same thing

fejiroofficial replied:
okay

Billmike commented on test/controller/questions.spec.js:
```diff
@@ -7,126 +7,110 @@ import { allQuestions, answers } from '../../datastore/questions';
+
+    describe('when invalid data is passed', () => {
+      it('should not create a question resource', (done) => {
+        const invalidQuestion = Object.assign({}, validQuestion, {
```
Same thing is applicable here:
```javascript
const invalidQuestion = { ...validQuestion, questionTitle: '', questionDescription: '' }
```

fejiroofficial replied:
noted

Billmike commented:
1. Your PR does not follow the laid out convention. The PR title doesn't conform to the standard.
2. You have 6 commits for this PR. You need to squash it into one commit. You can refer to [this](https://github.com/k88hudson/git-flight-rules)

fejiroofficial committed:
enhancing test coverage

fejiroofficial committed:
Implementing LFA feedback
This is a commit for implementing feedback.
1. Adding a new line to .slugignore
2. Using a spread operator instead of Object.assign
3. eslint extends airbnb for style guidance.

[Fixes]
